firedrake/plot.py

```diff
@@ -129,10 +128,10 @@ def bezier_plot(function, axes=None, **kwargs):
     M_inv = np.linalg.inv(M)
     cell_node_list = function.function_space().cell_node_list
     y_vals = np.dot(function.dat.data_ro[cell_node_list], M_inv)
-    x = SpatialCoordinate(function.function_space().mesh())
-    coords = Function(function.function_space())
+    x = SpatialCoordinate(mesh)
+    coords = Function(FunctionSpace(mesh, 'DG', deg))
     coords.interpolate(x[0])
-    x_vals = np.dot(coords.dat.data_ro[cell_node_list], M_inv)
+    x_vals = np.dot(coords.dat.data_ro[coords.function_space().cell_node_list], M_inv)
```

Comment:
To be honest, I am not sure about this. The assumption is that DG and CG have the same basis at the same given degree.